src/transformers/trainer.py

```diff
@@ -1243,7 +1243,10 @@ def _rotate_checkpoints(self, use_mtime=False) -> None:
             shutil.rmtree(checkpoint)
 
     def evaluate(
-        self, eval_dataset: Optional[Dataset] = None, ignore_keys: Optional[List[str]] = None
+        self,
+        eval_dataset: Optional[Dataset] = None,
+        ignore_keys: Optional[List[str]] = None,
+        metric_key_prefix: Optional[str] = "eval",
```

Comment:
```suggestion
        metric_key_prefix: str = "eval",
```

Reply:
cc https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint#:~:text=stands%20for%20any%20valid%20type,None%20%2C%20you%20should%20use%20Optional%20.&text=Note%20that%20there%20is%20technically,None%20to%20the%20Union%5B%5D%20.